misc/hooks.py

```diff
@@ -1080,13 +1305,29 @@ def make_ida_not_suck_cocks(nw_code):
     else:
         idaapi.__notification__.add(idaapi.NW_OPENIDB, database.config.__nw_init_info_structure__, -30)
 
+    ## keep track of individual tags like colors and type info
+    if idaapi.__version__ >= 7.2:
+        ui.hook.idb.add('item_color_changed', item_color_changed, 0)
+
+    if idaapi.__version__ >= 7.0:
+        ui.hook.idp.add('ev_init', typeinfo.database_init, 0)
+        ui.hook.idb.add('changing_ti', typeinfo.changing, 0)
+        ui.hook.idb.add('ti_changed', typeinfo.changed, 0)
+
+    # earlier versions of IDAPython don't expose anything about "extra" comments
+    # so we can't do anything here.
+    if idaapi.__version__ >= 6.9:
+        ui.hook.idb.add('extra_cmt_changed', extra_cmt.changed, 0)
+
     ## just some debugging notification hooks
     #[ ui.hook.ui.add(item, notify(item), -100) for item in ['range','idcstop','idcstart','suspend','resume','term','ready_to_run'] ]
     #[ ui.hook.idp.add(item, notify(item), -100) for item in ['ev_newfile','ev_oldfile','ev_init','ev_term','ev_newprc','ev_newasm','ev_auto_queue_empty'] ]
     #[ ui.hook.idb.add(item, notify(item), -100) for item in ['closebase','savebase','loader_finished', 'auto_empty', 'thunk_func_created','func_tail_appended'] ]
     #[ ui.hook.idp.add(item, notify(item), -100) for item in ['add_func','del_func','set_func_start','set_func_end'] ]
     #ui.hook.idb.add('allsegs_moved', notify('allsegs_moved'), -100)
     #[ ui.hook.idb.add(item, notify(item), -100) for item in ['cmt_changed', 'changing_cmt', 'range_cmt_changed', 'changing_range_cmt'] ]
+    #[ ui.hook.idb.add(item, notify(item), -100) for item in ['changing_ti', 'ti_changed', 'changing_op_type', 'op_type_changed'] ]
+    #[ ui.hook.idb.add(item, notify(item), -100op_type_changed for item in ['changing_op_ti', 'op_ti_changed'] ]
```

Comment:
This commented line includes a type-o and is also missing some other events that can use notify.